### DISCUSSION TITLE
How can I check the background color of the terminal?

### DISCUSSION THREAD

**Discussion by Zac-HD** [Textualize/rich#1170] [Q&A]
I'm using Rich to highlight output from [the Hypothesis ghostwriter](https://hypothesis.readthedocs.io/en/latest/ghostwriter.html) ([code](https://github.com/HypothesisWorks/hypothesis/blob/5f02ac4f5d4394ffe91db9b939276c8d4fd3ee82/hypothesis-python/src/hypothesis/extra/cli.py#L243-L252)), but unfortunately the `monokai` theme is unreadable on light-background terminals.  Is there any way to tell whether the terminal background color is light or dark, so that I can use `monokai_light` or `monokai` themes respectively?

Setting a background color leaves *many* trailing spaces when the output is piped into a file, as well as being less pretty (see #990).  The ansi_light or ansi_dark themes are readable (#284) but relatively ugly.

**Comment by willmcgugan** [comment]
There is no standard way to detect the terminal background. There are a few ways that work on some terminals, but on most terminals it is impossible AFAIK.

If you want the highlighting to blend with the user's color theme your only option is to use an ansi_* theme. Alternatively you could expose the theme via a env var or setting.

**Reply by Zac-HD** [reply]
Darn.  In the interests of simplicity I might just recommend that light-theme users run `NO_COLOR=1 hypothesis write ...` instead 😕

**Reply by willmcgugan** [reply]
If you like, but it would be easy enough to implement something like `LIGHT_BACKGROUND=1 hypothesis write`.

```python
syntax = Syntax(code, theme="fruity" if "LIGHT_BACKGROUND" in os.environ else "monokai")
```

**Reply by Zac-HD** [reply]
I ended up [going with the existing `DJANGO_COLORS` and `COLORFGBG` environment variables](https://github.com/HypothesisWorks/hypothesis/blob/ec884d88e9723c531e60c0282c518a2d4a4245dd/hypothesis-python/src/hypothesis/utils/terminal.py#L14-L36).

**Comment by erezsh** [comment]
Perhaps it's worth implementing something like this for rich? https://github.com/dalance/termbg

I think this issue is going to be quite common.
